src/puppetlabs/dujour/version_check.clj

```diff
@@ -88,10 +89,15 @@
                          (merge version-data)
                          json/generate-string)
         url update-server-url
-        {:keys [status body] :as resp} (client/post url
-                                         {:headers {"Accept" "application/json"}
-                                          :body request-body
-                                          :as :text})]
+        {:keys [status body] :as resp} (try
+                                         (client/post url
+                                                     {:headers {"Accept" "application/json"}
+                                                      :body request-body
+                                                      :as :text})
+                                         (catch IOException e
+                                           (throw+ {:kind ::connection-error
+                                                    :msg (.getMessage e)
+                                                    :exception e})))]
```

Comment:
Ah, that's the standard way to craft exceptions in puppet enterprise:

https://github.com/puppetlabs/puppet-nogui/blob/master/patterns/api_style_guide.md#errors